src/amo/components/App/index.js

```diff
@@ -49,8 +45,13 @@ interface MozNavigator extends Navigator {
   mozAddonManager?: Object;
 }
 
+export const isHomePage = (location: ReactRouterLocationType) => {
+  return location.pathname.split('/').filter(Boolean).length === 2;
```

Comment:
This seems pretty fragile and based on a lot of assumptions. 

Can you at least use a shared variable like `parts = ['lang', 'application']` that will be used to [create the route](https://github.com/mozilla/addons-frontend/blob/master/src/amo/components/Routes/index.js#L74)?

Reply:
>
> This seems pretty fragile and based on a lot of assumptions.
>

It's based on *one* assumption, which is that the home page has two parts in its path, and any non home page will have more than 2. I suppose it should have been `location.pathname.split('/').filter(Boolean).length <= 2`.
 
> Can you at least use a shared variable like `parts = ['lang', 'application']` that will be used to [create the route](https://github.com/mozilla/addons-frontend/blob/master/src/amo/components/Routes/index.js#L74) [](/mozilla/addons-frontend/blob/HEAD@{2019-10-03T20:40:39Z}/src/amo/components/Routes/index.js#L74)?

Is the suggestion that I should use `match` to check the params extracted by `react-router-dom` and verify that both `lang` and `application` exist (and nothing else does)? I could do that as an extra check, but I'm not sure there's much value in the former, as all pages should have a `lang` and an `application`.

Reply:
> > This seems pretty fragile and based on a lot of assumptions.
> 
> It's based on _one_ assumption, which is that the home page has two parts in its path, and any non home page will have more than 2. I suppose it should have been `location.pathname.split('/').filter(Boolean).length <= 2`.
> 
> > Can you at least use a shared variable like `parts = ['lang', 'application']` that will be used to [create the route](https://github.com/mozilla/addons-frontend/blob/master/src/amo/components/Routes/index.js#L74) ?
> 
> Is the suggestion that I should use `match` to check the params extracted by `react-router-dom` and verify that both `lang` and `application` exist (and nothing else does)? I could do that as an extra check, but I'm not sure there's much value in the former, as all pages should have a `lang` and an `application`.

Not all routes will have a lang and application. See https://github.com/mozilla/addons-frontend/blob/d1b016dcfaeaf7a4bf4ec8acd6b915aa0d34214c/config/default-amo.js#L63-L84